api/hackerProfile.js

```diff
@@ -145,6 +145,29 @@ router.put("/", async (req, res) => {
   return res.json({ hackerProfile: updatedHackerProfile });
 });
 
+// Direct patch the referrerCode if there isn't one
+router.put("/referrerCode", async (req, res) => {
```

Comment:
add `utils.authMiddleware`, since otherwise people with null sessions can hit this endpoint.